main_test.go

```diff
@@ -18,3 +18,19 @@ func TestTimeGranularity(t *testing.T) {
 	assert.Equal(t, start, time.Date(2017, 7, 10, 0, 0, 0, 0, time.UTC))
 	assert.Equal(t, end, time.Date(2017, 7, 17, 0, 0, 0, 0, time.UTC))
 }
+
+func TestIsInputDataOlder(t *testing.T) {
+	inputDataDate, _ := time.Parse(time.RFC3339, "2017-08-15T14:00:00Z")
+	targetDataDate, _ := time.Parse(time.RFC3339, "2017-08-15T21:00:00Z")
+
+	assert.Equal(t, false, isInputDataOlder(inputDataDate, &targetDataDate, "day", false))
+	assert.Equal(t, true, isInputDataOlder(inputDataDate, &targetDataDate, "hour", false))
+	assert.Equal(t, false, isInputDataOlder(inputDataDate, nil, "hour", false))
+
+	// Simulate Redshift timestamp without time zones that is parsed as UTC but is actually PST
```

Comment:
Good idea! This seems like the most important functionality for us to test as well - 11365a7